src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1429,39 +1430,59 @@ export class ResultsViewPageStore {
         }
     });
 
-    public getPutativeDriverInfo(mutation:Mutation) {
-        const oncoKbDatum:IndicatorQueryResp | undefined | null | false = this.mutationAnnotationSettings.oncoKb &&
-            this.getOncoKbMutationAnnotation.isComplete &&
-            this.getOncoKbMutationAnnotation.result(mutation);
-
-        let oncoKb:string = ((oncoKbDatum && oncoKbDatum.oncogenic) || "").toLowerCase();
-        if (ONCOKB_ONCOGENIC_LOWERCASE.indexOf(oncoKb) === -1) {
-            oncoKb = "";
-        }
-
-        const hotspots:boolean =
-            (this.mutationAnnotationSettings.hotspots &&
-            this.isHotspot.isComplete &&
-            this.isHotspot.result(mutation));
-
-        const cbioportalCount:boolean =
-            (this.mutationAnnotationSettings.cbioportalCount &&
-            this.getCBioportalCount.isComplete &&
-            this.getCBioportalCount.result!(mutation) >=
-            this.mutationAnnotationSettings.cbioportalCountThreshold);
-
-        const cosmicCount:boolean =
-            (this.mutationAnnotationSettings.cosmicCount &&
-            this.getCosmicCount.isComplete &&
-            this.getCosmicCount.result!(mutation) >= this.mutationAnnotationSettings.cosmicCountThreshold);
-
-        return {
-            oncoKb,
-            hotspots,
-            cbioportalCount,
-            cosmicCount
-        };
-    }
+    readonly getPutativeDriverInfo = remoteData({
+        await:()=>{
+            const toAwait = [];
```

Comment:
dynamic `await`... this is necessary because otherwise we're querying for, and waiting for, data before we need it